WordPress/Classes/ViewRelated/Post/EditPostViewController.swift

```diff
@@ -0,0 +1,293 @@
+        return .LightContent
+    }
+
+    // MARK: show the editor
+
+    private func showEditor() {
+        let editorSettings = EditorSettings()
+        let editor: UIViewController
+        if editorSettings.visualEditorEnabled {
+            if editorSettings.nativeEditorEnabled {
+                editor = editPostInNativeEditor()
+            } else {
+                editor = editPostInNewEditor()
+            }
+        } else {
+            editor = editPostInOldEditor()
+        }
+
+        postPost.presentViewController(editor, animated: !showImmediately) {
+            let generator = WPImpactFeedbackGenerator(style: .Medium)
+            generator.impactOccurred()
+        }
+    }
+
+    private func editPostInNativeEditor() -> UIViewController {
+        let postToEdit: Post
+        if let post = self.post {
+            postToEdit = post
+        } else {
+            let context = ContextManager.sharedInstance().mainContext
+            let postService = PostService(managedObjectContext: context)
+            postToEdit = postService.createDraftPostForBlog(blog)
+            WPAppAnalytics.track(.EditorCreatedPost, withProperties: ["tap_source": "posts_view"], withBlog: blog)
+        }
+
+        let postViewController = AztecPostViewController(post: postToEdit)
+        let navController = UINavigationController(rootViewController: postViewController)
+        navController.modalPresentationStyle = .FullScreen
+        return navController
+    }
+
+    private func editPostInNewEditor() -> UIViewController {
+        let targetPost: Post
+        if let post = post {
+            targetPost = post
+        } else {
+            let context = ContextManager.sharedInstance().mainContext
+            let postService = PostService(managedObjectContext: context)
+            targetPost = postService.createDraftPostForBlog(blog)
+            post = targetPost
+            WPAppAnalytics.track(.EditorCreatedPost, withProperties: ["tap_source": "posts_view"], withBlog: blog)
+        }
+        let postViewController = WPPostViewController(post: targetPost, mode: kWPPostViewControllerModeEdit)
+        postViewController.isOpenedDirectlyForPhotoPost = openWithMediaPicker
+
+        postViewController.onClose = { [weak self] (viewController, changesSaved) in
+            self?.closeEditor(changesSaved)
+        }
+
+        let navController = UINavigationController(rootViewController: postViewController)
+        navController.restorationIdentifier = WPEditorNavigationRestorationID
+        navController.restorationClass = WPPostViewController.self
+        navController.toolbarHidden = false // Fixes incorrect toolbar animation.
+        navController.modalPresentationStyle = .FullScreen
+
+        return navController
+    }
+
+    private func editPostInOldEditor() -> UIViewController {
+        let editPostViewController: WPLegacyEditPostViewController
+
+        let targetPost: Post
+        if let post = post {
```

Comment:
This section of code to unwrap the existing post or create a new one seems to be repeated in all three of these `editIn...` methods. Can we pull it out to a helper method?